specification/machinelearningservices/data-plane/Microsoft.MachineLearningServices/preview/2019-05-01/execution.json

```diff
@@ -0,0 +1,1103 @@
+          "type": "string",
+          "example": "FindSquaresJob"
+        },
+        "maxRunDurationSeconds": {
+          "format": "int64",
+          "description": "Maximum allowed time for the run. The system will attempt to automatically cancel the run if it took longer than this value.\r\nMaxRunDurationSeconds=null means infinite duration.",
+          "type": "integer",
+          "example": "84000"
+        },
+        "nodeCount": {
+          "format": "int32",
+          "description": "Number of compute nodes to run the job on. Only applies to AMLCompute.",
+          "type": "integer",
+          "example": "1"
+        },
+        "environment": {
+          "$ref": "#/definitions/EnvironmentDefinition",
+          "description": "The environment definition, This field configures the python environment.\r\nIt can be configured to use an existing Python environment or configured to setup a temp environment for the experiment.\r\nThe definition is also responsible for setting the required application dependencies."
+        },
+        "history": {
+          "$ref": "#/definitions/HistoryConfiguration",
+          "description": "This section is used to disable and enable experiment history logging features."
+        },
+        "spark": {
+          "$ref": "#/definitions/SparkConfiguration",
+          "description": "When the platform is set to Pyspark, The spark configuration is used to set the default sparkconf for the submitted job."
+        },
+        "amlCompute": {
+          "$ref": "#/definitions/AMLComputeConfiguration",
+          "description": "The attribute is used to configure details of the compute target to be created during experiment.\r\nThe configuration only takes effect when the target is set to \"amlcompute\"."
+        },
+        "tensorflow": {
+          "$ref": "#/definitions/TensorflowConfiguration",
+          "description": "The attribute is used to configure the distributed tensorflow parameters.\r\nThis attribute takes effect only when the framework is set to TensorFlow, and the communicator to ParameterServer.\r\nAmlCompute is the only supported compute for this configuration."
+        },
+        "mpi": {
+          "$ref": "#/definitions/MpiConfiguration",
+          "description": "The attribute is used to configure the distributed MPI job parameters.\r\nThis attribute takes effect only when the framework is set to Python, and the communicator to OpenMpi or IntelMpi.\r\nAmlCompute is the only supported compute type for this configuration."
+        },
+        "hdi": {
+          "$ref": "#/definitions/HdiConfiguration",
+          "description": "This attribute takes effect only when the target is set to an Azure HDI compute.\r\nThe HDI Configuration is used to set the YARN deployment mode. It is defaulted to cluster mode."
+        },
+        "containerInstance": {
+          "$ref": "#/definitions/ContainerInstanceConfiguration"
+        },
+        "exposedPorts": {
```

Comment:
exposedPorts should not be documented; it's a legacy option that will likely be removed. It's not something we're guaranteeing backward-compatibility for.

Reply:
done